src/components/validation/resources/http.rs

```diff
@@ -257,13 +267,26 @@ fn spawn_output_http_server(
     // First, we'll build and spawn our HTTP server.
     let decoder = codec.into_decoder()?;
 
+    // the number of events we expect to receive from the sink.
+    // any happy path events + any events that are flagged as external resource should reject.
+    let expected_output_events = input_events
+        .iter()
+        .filter(|te| !te.should_fail() || te.should_reject())
+        .count();
+
+    let should_reject = input_events.iter().filter(|te| te.should_reject()).count() > 0;
+
+    let (server_sent_all_tx, mut server_sent_all_rx) = mpsc::channel(1);
+
     let (_, http_server_shutdown_tx) = spawn_http_server(
         task_coordinator,
         &config,
         runner_metrics,
         move |request, output_runner_metrics| {
             let output_tx = output_tx.clone();
             let mut decoder = decoder.clone();
+            let mut decoded_events = 0;
+            let server_sent_all_tx = server_sent_all_tx.clone();
```

Comment:
For what it's worth, this is only going to work on a per-connection basis.

If you ever ended up with a sink that, for example, gets N events in (where you expected to receive all N of those events here) but sends them across M requests (M > N), then this logic wouldn't ever get triggered, I don't think?

Like it might work now based on how things are batched, but it _feels_ a little shaky. I think it might just be easier to perhaps send the actual count of events processed over the channel, and tally those counts up down near `if expected_output_events > 0 {`... then you'd stop trying to receive once you've hit the intended received event count.

Reply:
Yes indeed. I did run into this on my branch based from this branch for the splunk_hec logs sink.

It has a couple other changes in the framework (including the one you identified here). So I'll probably need to split that out for a review as well. I'll address it in that PR since I already have changes in that branch.

Reply:
This is resolved now that the output driver itself is tracking the event expectations